aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -268,7 +311,28 @@ int64_t hsum(const int32_t* A, int len) {
   for (const auto k : c10::irange(4)) {
     row_sum += temp[k];
   }
-#endif // CPU_CAPABILITY_AVX2
+#elif defined(CPU_CAPABILITY_AVX512)
+  __m512i sum_epi64 = _mm512_setzero_si512();
+  // vectorized
+  for (; i < len / 16 * 16; i += 16) {
+    __m512i src_epi32 = _mm512_loadu_si512(reinterpret_cast<__m512i const*>(A + i));
+    // widen
+    __m256i src_lo_epi32 = _mm512_castsi512_si256(src_epi32);
+    __m256i src_hi_epi32 =
+        _mm256_castps_si256(_mm512_extractf32x8_ps(_mm512_castsi512_ps(src_epi32), 1));
```

Comment:
lol

Reply:
`_mm256_extractf128_si256` in AVX2 doesn't have an AVX512 counterpart, so this line looks awkward. The casts are no-ops.

However, I had actually been more inclined to simply use `_mm512_extracti32x8_epi32`, as such ops should simply extract a group of bits, so float/int types shouldn't matter here.

Reply:
I'm wondering if I should also change the `_mm256_extractf128_si256` to `_mm256_extracti128_si256` in the corresponding AVX2 code.